src/processor/tree/tree_wrapper.rs

```diff
@@ -8,12 +8,21 @@ use state_reconstruct_fetcher::{
     database::InnerDB,
     types::{v2::PackingType, CommitBlock},
 };
+use thiserror::Error;
 use tokio::sync::Mutex;
 use zksync_merkle_tree::{Database, MerkleTree, RocksDBWrapper, TreeEntry};
 
 use super::RootHash;
 
+#[derive(Error, Debug)]
+pub enum TreeError {
+    #[error("block mismatch")]
+    BlockMismatch,
+}
+
 pub struct TreeWrapper {
+    index2key: HashMap<u64, U256>,
+    key2value: HashMap<U256, H256>,
```

Comment:
Would it be possible to leverage the `zkSyncTree` struct instead? [This method](https://github.com/matter-labs/zksync-era/blob/6f63c53549fda7e253e1d4cb4ed42ec6a2e3e9f1/core/lib/merkle_tree/src/domain.rs#L382-L386), in particular, seems like just the thing we need.

Reply:
as I mentioned on Slack, that cannot access `MerkleTree::latest_version()`...

Reply:
That's true, but in its stead we should be able to use [zkSyncTree::next_l1_batch_number](https://github.com/matter-labs/zksync-era/blob/add2321e9f039afbbb4e90d1fb0269053608af9e/core/lib/merkle_tree/src/domain.rs#L153-L160), that calls it under the hood.

Reply:
not quite - the returned value is off by 1, and irritatingly enough we cannot modify it, because the `L1BatchNumber` type isn't accessible from outside the package...